PyExpLabSys/drivers/keithley_2400.py

```diff
@@ -128,65 +131,99 @@ def read_voltage(self):
         # Values is: voltage, current, ohm, time, status
         # Only the voltage is measured, current is either
         # NaN or the source-setpoint.
+        print('DEBUG KLAF 2400 Voltage', repr(raw))
         values = raw.split(',')
         voltage = float(values[0])
         # timestamp = float(values[3])
         # print(self._parse_status(values[4]))
         # Also return timestamp?
         return voltage
 
-    def set_source_function(self, function=None):
-        if function in ('i', 'I'):
-            self.scpi_comm('SOURCE:FUNCTION CURRENT')
-        if function in ('v', 'V'):
-            self.scpi_comm('SOURCE:FUNCTION VOLTAGE')
+    def set_source_function(self, function=None, source_range=None):
```

Comment:
Would be nice with the source_range format (allowed values) in the method string.